### ISSUE TITLE
Bug copy-paste Inline Classes

### ISSUE THREAD

**Issue by AndreaGProg** [froala/wysiwyg-editor#3809]
Bug copy-paste Inline Classes:

1 goto https://www.froala.com/wysiwyg-editor/examples/inline-classes
2 add a class to a word
3 copy all sentence
4 paste -> class not copied!

If copy-and-paste only the single sticky word, you might think that it works, but in reality the class has been transformed into a style (class="myCls" to style="....") - bad practice.

**Comment by AndreaGProg** [comment]
![bug-froala-inlineclass](https://user-images.githubusercontent.com/60134321/73340904-de86ac80-427b-11ea-83a0-c53dc116dc59.jpg)

**Comment by AndreaGProg** [comment]
https://www.froala.com/wysiwyg-editor/examples/inline-classes

**Feature request:**

**1) Ability to add multiple "inlineClasses" buttons** 
(without having to implement them with the FroalaEditor.RegisterCommand)

**2) Improve the code**

a) add a class to a word:
`<span class="fr-class-code">dropdown</span>`
b) add another class to the same word:
`<span class="fr-class-highlighted"><span class="fr-class-code">dropdown</span></span>`
c) could be optimized like this:
`<span class="fr-class-highlighted fr-class-code">dropdown</span>`

OR

a) add class: `<span class="fr-class-code">dropdown</span>`
b) remove class: `&#8203;desired&#8203;`
c) the " &#8203 " symbols should be removed

OR

a) add class to all sentence:
`<p><span class="fr-class-highlighted">Use the Inline Class dropdown to select the desired inline class for the selected text inside the rich text editor.</span></p>`
b) could be optimized like this:
`<p class="fr-class-highlighted">Use the Inline Class dropdown to select the desired inline class for the selected text inside the rich text editor.</p>`

**Comment by ilyaskarim** [comment]
@casualuser please see https://jsfiddle.net/ospyuvzb/, You will see highlighted which written as <p><span class="fr-class-highlighted">How are you?</span></p>, After copying a single word from it and pasting it generates new styles. Do you think this is a bug?

**Comment by AndreaGProg** [comment]
Yes, I think if a Class is present in the "inlineClasses" property, it must remain like this.

For example, later on, I might want to run other javascript code on that Class.
Or a Class could be placed in a Media Query, while a Style could not.

And I think it's always better to work with the Class and not with Style.
I think that transforming a Class into a Style is bad code.

What do you think about it?

**Comment by AndreaGProg** [comment]
It would be nice to be able to add more custom buttons

From 
```
inlineClasses: {
	'fr-class-code': 'Code',
	'fr-class-highlighted': 'Highlighted',
	'fr-class-transparency': 'Transparent'
}
```

To
```
inlineClasses: [
{
	name: 'Inline Class',
	icon: '/my/icon.jpg',
	options: {
		'fr-class-code': 'Code',
		'fr-class-highlighted': 'Highlighted',
		'fr-class-transparency': 'Transparent'
	}
},
{
	name: 'My Name',
	icon: '/my/image.jpg',
	options: {
		'fr-class-aaa': 'AAA',
		'fr-class-bbb': 'BBB'
	}
}
]
```

**Comment by AndreaGProg** [comment]
New bug:
1 https://jsfiddle.net/ospyuvzb/
2 remove "highlighted" from the single word "are"
3 copy all
4 paste -> bug -> all sentence is "highlighted"

![new-bug](https://user-images.githubusercontent.com/60134321/74547361-49a6d300-4f4c-11ea-9b28-d3c992d7d343.jpg)

**Comment by AndreaGProg** [comment]
New new bug:
1 https://jsfiddle.net/ospyuvzb/
2 remove "highlighted" from the single word "are"
3 copy all
4 paste -> bug -> all sentence is "highlighted"
5 copy all 
6 paste -> bug -> the new two lines do not have "highlighted"

<img width="339" alt="Schermata 2020-02-14 alle 17 52 38" src="https://user-images.githubusercontent.com/60134321/74551071-d3f23580-4f52-11ea-9497-c916535e2053.png">

**Comment by ilyaskarim** [comment]
For the first one, please see https://jsfiddle.net/L5qg82n1/3/. You need to use useClasses option as true so it won't make style attribute upon paste. Please see https://froala.com/wysiwyg-editor/docs/options#useClasses.

**Comment by ilyaskarim** [comment]
For the second one, Please see jsFiddle https://jsfiddle.net/5h0jexym/1/ I have confirmed the bug.

**Comment by ilyaskarim** [comment]
I'll ask @casualuser to mark this as bug.

**Comment by AndreaGProg** [comment]
> For the first one, please see https://jsfiddle.net/L5qg82n1/3/. You need to use useClasses option as true so it won't make style attribute upon paste. Please see https://froala.com/wysiwyg-editor/docs/options#useClasses.

Thanks for this valid solution.

Maybe you can make this system dynamic without having to set this property.
When copying, save both the class and the style.
When pasting, check if the class exists in the inlineClasses list.
If it exists in the list, paste the class and not the style, otherwise paste the style is not the class.
To do this in the clipboard we will save three values: the class name, the style generated by the class, and if exists the style attribute (this third parameter is useful if we have both a class and a style - like this: < p class="..." style="..." >).

**Comment by claudchan-alyka** [comment]
Hi there, 
can we also add support bind to the specific element tag?
Related issue: #4680 https://github.com/froala/wysiwyg-editor/issues/4680

**Comment by ilyaskarim** [comment]
We’re closing this issue due to inactivity. If you’re still experiencing this problem on the latest version of our editor, please feel free to reopen the issue and let us know. You can find details on the latest updates here: [Froala Editor Changelog](https://froala.com/wysiwyg-editor/changelog/).
